### PR TITLE
2.0.0

### LINKED ISSUE #4: Flexbox
We want to transition this grid implementation to use flexbox by default and fallback to `inline-block` where flexbox is not supported. This should give us an even more powerful and flexible grid solution whilst offering a few new cool features.

I've got a quick spike of what this might look like here: http://codepen.io/shaunbent/pen/obrBxb?editors=1100

### LINKED ISSUE #5: Integrate CI
We've done this for Typography so should probably do the same for the Grid.

### LINKED ISSUE #6: Grid Bookmarklet
We should bundle the Grid Bookmarklets as part of this repository:

https://confluence.dev.bbc.co.uk/display/~andrew.jones3/Fully+Responsive+Bookmarklets

### LINKED ISSUE #7: RTL Support
News/World Service need the ability to support RTL text and layouts.

We can use `gel-layout--rev` to reverse the layouts of the grid but this includes a forced `text-align`.

We need to introduce a new modifier class of `gel-layout--rtl` which applies: `direction: rtl;` and applies `text-align: start;` to the `gel-layout__item`s.

Here is a codepen of this in action: http://codepen.io/shaunbent/pen/MyqKBQ

### LINKED ISSUE #10: Incorrect 1280 breakpoints
We have the 1008px (`@xl`) breakpoint listed as a breakpoint which only applies when we're using a 1280px grid.

I think this is a bug. If we apply an adjustment at this breakpoint we'd want it to apply regardless of if the next 1280px (`@xxl`) breakpoint is enabled or not.

As the `@xl` breakpoint is currently listed as a 1280px breakpoint it gets scoped in the `b-pw-1280` class so these styles don't apply on non 1280 pages, when I think they should really.

cc: @wildlyinaccurate

### PR DESCRIPTION
**Pull request by shaunbent**
### Flexbox

Implementing a flexbox grid by default falling backing to our existing `inline-block` grid. Resolve #4 

- Add flexbox
- Disables autoprefixer within the grid as we want fine grain control of which vendor prefixes are output
- Create new `--equal` modifier to enable equal height columns
- Create new `--fit` modifier to enable automatically sizing grids
- Create new independent alignment classes: `.gel-layout__item--top`,  `.gel-layout__item--middle`, `.gel-layout__item--bottom`

### Right-to-Left Support

- Add Right-to-Left support. Resolves #7 
- Pulls in latest [GEL Sass Tools](https://github.com/bbc/gel-sass-tools/tree/1.2.0) with RTL tools

### Demo

- Adds a fallback colour to the grid demo
- Enables `inline-block` whitespace fix for grid demo
- Add a Right-to-Left toggle option to the demo
- Enable Right-to-Left support for the demo

### Misc

- Removes the `xl` breakpoint from within the 1280px scoping class - resolves #10 
- Removes some mixin arguments and associated logic to simplify things
- Removes extend placeholders, `@extend`s are 💩 
- Integrate [Travis CI](https://travis-ci.org/bbc/gel-grid/) - resolves #5 
- Add links to Grid Overlay Bookmarklets - resolves #6

### CODE REVIEW

**Comment by howieweiner** [comment]
So, does this mean that gel-grid will no longer support IE8? If so, this is a breaking change, so I'd suggest a major version bump and clarification in the docs.

**Comment by shaunbent** [comment]
We will maintain the same level of browser support. Where `flexbox` is not supported we will fallback to our existing `inline-block` grid which supports IE8.

I am in the process of testing this across as wide a range of devices as I can to we can understand what gets what in terms of grid implementation.

There will be some new grid features that will only work for flexbox supporting browsers, such as equal height columns, automatic grids and alignment of individual items.

I was thinking of going with a breakpoint change anyway to allow people to opt into this.

**Comment by howieweiner** [comment]
Cool. I deleted my original comment as I now realise you have fallback support

File: demo/src/assets/sass/_utilities.float.scss [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    .gs-u-fl#{$breakpoint-suffix} {
+        float: left !important;
+    }
+
+    .gs-u-fr#{$breakpoint-suffix} {
+        float: right !important;
+    }
+
+    .gs-u-fn#{$breakpoint-suffix} {
+        flot: none !important;
```
Comment (wildlyinaccurate) [inline demo/src/assets/sass/_utilities.float.scss]:
```
F L O T
L
O
T
```

**Pull request merged**